src/char/char_clif.cpp

```diff
@@ -816,6 +820,165 @@ void chclif_send_map_data( int fd, std::shared_ptr<struct mmo_charstatus> cd, ui
+	char_set_char_online( -2, char_id, sd->account_id );
+
+	struct mmo_charstatus char_dat;
+
+	if( !char_mmo_char_fromsql( char_id, &char_dat, true ) ) {
+		/* failed? set it back offline */
+		char_set_char_offline( char_id, sd->account_id );
+		/* failed to load something. REJECT! */
+		chclif_reject( fd, 0 ); /* jump off this boat */
+		return 1;
+	}
+
+	// Have to switch over to the DB instance otherwise data won't propagate [Kevin]
+	std::shared_ptr<struct mmo_charstatus> cd = util::umap_find( char_get_chardb(), char_id );
+
+	if( charserv_config.log_char ){
+		char esc_name[NAME_LENGTH*2+1];
+
+		Sql_EscapeStringLen( sql_handle, esc_name, char_dat.name, strnlen( char_dat.name, NAME_LENGTH ) );
+
+		if( SQL_ERROR == Sql_Query( sql_handle, "INSERT INTO `%s`(`time`, `account_id`,`char_num`,`name`) VALUES (NOW(), '%d', '%d', '%s')", schema_config.charlog_db, sd->account_id, p.slot, esc_name ) ){
+			Sql_ShowDebug( sql_handle );
+		}
+	}
+
+	ShowInfo( "Selected char: (Account %d: %d - %s)\n", sd->account_id, p.slot, char_dat.name );
+
+	// Check if there is really no mapserver for the last point where the player was
+	int32 mapserver = char_search_mapserver( cd->last_point.map, -1, -1 );
+
+	// It was not an unavailable map
+	if( mapserver >= 0 ){
+		chclif_reject( fd, 0 ); // rejected from server
+		return 1;
+	}
+
+	if( static_cast<size_t>( p.mapnumber ) >= accessible_maps.size() ){
+		chclif_reject( fd, 0 ); // rejected from server
+		return 1;
+	}
+
+	s_point_str& accessible_map = accessible_maps[p.mapnumber];
+
+	safestrncpy( cd->last_point.map, accessible_map.map, sizeof( cd->last_point.map ) );
+	cd->last_point.x = accessible_map.x;
+	cd->last_point.y = accessible_map.y;
+
+	mapserver = char_search_mapserver( cd->last_point.map, -1, -1 );
+
+	// No mapserver found for our accessible map
+	if( mapserver < 0 ){
+		chclif_reject( fd, 0 ); // rejected from server
+		return 1;
+	}
+
+	int map_fd;
+
+	// Send NEW auth packet [Kevin]
+	// FIXME: is this case even possible? [ultramage]
+	if( ( map_fd = map_server[mapserver].fd ) < 1 || session[map_fd] == nullptr ){
+		ShowError( "parse_char: Attempting to write to invalid session %d! Map Server #%d disconnected.\n", map_fd, mapserver );
+		map_server[mapserver].fd = -1;
+		memset( &map_server[mapserver], 0, sizeof( struct mmo_map_server ) );
+		chclif_send_auth_result( fd, 1 ); // Send server closed.
+		return 1;
+	}
+
+	chclif_send_map_data( fd, cd, ipl, mapserver );
+
+	// create temporary auth entry
+	std::shared_ptr<struct auth_node> node = std::make_shared<struct auth_node>();
+
+	node->account_id = sd->account_id;
+	node->char_id = cd->char_id;
+	node->login_id1 = sd->login_id1;
+	node->login_id2 = sd->login_id2;
+	node->sex = sd->sex;
+	node->expiration_time = sd->expiration_time;
+	node->group_id = sd->group_id;
+	node->ip = ipl;
+
+	char_get_authdb()[node->account_id] = node;
```

Comment:
Lets move this instead of assignment

Reply:
As discussed in Discord this should be dealt with in a broader scope.
But good point!